lib/IRGen/IRGenSIL.cpp

```diff
@@ -1932,6 +1936,25 @@ static llvm::Value *createStringValAddr(IRGenModule &IGM, StringRef strVal) {
   return opNameValAddr;
 }
 
+template <typename SrcT>
+void IRGenSILFunction::createArrayAndSize(
+    ArrayRef<SrcT> elements, bool isFloat, const char *arrayName,
+    std::function<llvm::Value *(SrcT elt)> convertElt, Address *array,
+    llvm::Value **numElts) {
+  *numElts = llvm::ConstantInt::get(IGM.Int32Ty, elements.size());
+  *array = isFloat ? createAlloca(IGM.FloatTy, *numElts,
```

Comment:
The two `createAlloca`s are identical except the type here. Consider a simpler approach:
```cpp
auto *type = isFloat ? IGM.FloatTy : (llvm::Type *)IGM.Int64Ty;
*array = createAlloca(type, *numElts, IGM.getPointerAlignment(), "tensorEltVals");
```

Reply:
Good idea! Done.